File: src/test/java/kitchenpos/FakeOrderTableRepository.java
package kitchenpos;

import kitchenpos.domain.OrderTable;
import kitchenpos.domain.OrderTableRepository;

import java.util.*;

public class FakeOrderTableRepository implements OrderTableRepository {

    Map<UUID, OrderTable> orderTables = new HashMap<>();


    @Override
    public Optional<OrderTable> findById(UUID orderTableId) {
        return Optional.ofNullable(orderTables.get(orderTableId));
    }

    @Override
    public OrderTable save(OrderTable orderTable) {
        orderTables.put(orderTable.getId(), orderTable);
        return orderTable;
    }

    @Override
    public List<OrderTable> findAll() {
        return new ArrayList<>(orderTables.values());
    }
}

File: src/test/java/kitchenpos/testsupport/FakeOrderTableRepository.kt
package kitchenpos.testsupport

import java.util.Optional
import java.util.UUID
import java.util.concurrent.ConcurrentHashMap
import kitchenpos.domain.OrderTable
import kitchenpos.domain.OrderTableRepository

class FakeOrderTableRepository : OrderTableRepository {
    private val storage: MutableMap<UUID, OrderTable> = ConcurrentHashMap()

    override fun findById(orderTableId: UUID): Optional<OrderTable> {
        return Optional.ofNullable(storage[orderTableId])
    }

    override fun save(orderTable: OrderTable): OrderTable {
        storage[orderTable.id] = orderTable
        return orderTable
    }

    override fun findAll(): MutableList<OrderTable> {
        return storage.values.toMutableList()
    }
}